### ISSUE TITLE
Download YouTube playlist videos in a different way

### ISSUE THREAD

**Issue by ibrahimk157** [Just-Some-Bots/MusicBot#2094]
Please tick all applicable boxes.

- [X] I am using Python 3.5.3 or higher (run `python --version` on the command line)
- [X] I have followed the [official guides](https://github.com/Just-Some-Bots/MusicBot/wiki) to install the bot for my system
- [X] I have updated my dependencies to the latest version using the appropriate `update` script

Which version are you using?

- [ ] The latest `master` version (release-260419)
- [X] The latest `review` version

What type of issue are you creating?

- [X] Bug
- [ ] Feature request
- [ ] Question

----

### Description of issue

When the bot is commanded to play a YouTube playlist, it downloads all the videos of the playlist all at once... which in a large playlist makes YouTube IP block the VPS the bot is running on.

This behavior doesn’t make sense!

Why doesn’t the bot instead only download the first song, plays it, when it’s done it downloads the second song, plays it, etc...

Much better than downloading all the songs of a playlist at once! And will for sure prevent YouTube’s annoying IP block.

### Steps to reproduce

1. Make it play any YouTube playlist with many videos (example: https://www.youtube.com/playlist?list=PL4o29bINVT4EG_y-k5jGoOu3-Am8Nvi10)

2. It will try to download all of the videos all at once and will get IP blocked by Youtube in no time.

----

### Log file
Please attach your MusicBot log file (located at `logs/musicbot.log`) to this issue. You can do so by dragging and dropping the file here. **If you do not include your log file, you WILL be asked to provide one.**

```[29.976129532] 2020-07-14 21:31:48,849 - INFO - launcher: Starting sanity checks
[30.515432358] 2020-07-14 21:31:48,849 - INFO - launcher: Checking for Python 3.5+
[30.718088150] 2020-07-14 21:31:48,849 - INFO - launcher: Checking console encoding
[30.900955200] 2020-07-14 21:31:48,849 - INFO - launcher: Ensuring we're in the right environment
[448.570251465] 2020-07-14 21:31:49,267 - INFO - launcher: Required checks passed.
[449.181795120] 2020-07-14 21:31:49,268 - INFO - launcher: Optional checks passed.
[449.469566345] 2020-07-14 21:31:49,268 - INFO - launcher: Moving old musicbot log

######################### PRE-RUN SANITY CHECKS PASSED #########################

[1.0954689979553223] 2020-07-14 21:31:49,914 - WARNING - musicbot.config | In config.py::MainThread(140250957956928), line 123 in run_checks: i18n file does not exist. Trying to fallback to config/i18n/en.json.
[1.0958342552185059] 2020-07-14 21:31:49,914 - INFO - musicbot.config | In config.py::MainThread(140250957956928), line 134 in run_checks: Using i18n: config/i18n/en.json
[1.0980954170227051] 2020-07-14 21:31:49,917 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 239 in _setup_logging: Set logging level to INFO
[1.1010162830352783] 2020-07-14 21:31:49,920 - DEBUG - musicbot.json | In json.py::MainThread(140250957956928), line 8 in __init__: Init JSON obj with config/i18n/en.json
[1.1118435859680176] 2020-07-14 21:31:49,930 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 90 in __init__: Starting MusicBot release-260819-72-g2350384
[1.1124629974365234] 2020-07-14 21:31:49,931 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 96 in __init__: Loaded autoplaylist with 2531 entries
[1.1938691139221191] 2020-07-14 21:31:50,012 - DEBUG - musicbot.spotify | In spotify.py::MainThread(140250957956928), line 81 in get_token: Created a new access token: {'access_token': 'BQBYcGaTU42K96Gnt82HevLsnJEwJOVWAES6nNenj54CuwRTc6jCZH0ua598Lsd790dqgTGSY1IefZahBno', 'token_type': 'Bearer', 'expires_in': 3600, 'scope': '', 'expires_at': 1594765910}
[1.1942601203918457] 2020-07-14 21:31:50,013 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 121 in __init__: Authenticated with Spotify successfully using client ID and secret.
[3.6479089260101318] 2020-07-14 21:31:52,466 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 965 in on_ready: Connection established, ready to go.
[3.6483943462371826] 2020-07-14 21:31:52,467 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 356 in _cache_app_info: Caching app info
[3.7539911270141602] 2020-07-14 21:31:52,573 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 787 in _scheck_ensure_env: Ensuring data folders exist
[3.7552502155303955] 2020-07-14 21:31:52,574 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 803 in _scheck_server_permissions: Checking server permissions
[3.7554564476013184] 2020-07-14 21:31:52,574 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 807 in _scheck_autoplaylist: Auditing autoplaylist
[3.7556393146514893] 2020-07-14 21:31:52,574 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 811 in _scheck_configs: Validating config
[3.7558276653289795] 2020-07-14 21:31:52,574 - DEBUG - musicbot.config | In config.py::MainThread(140250957956928), line 237 in async_validate: Validating options...
[3.7559976577758789] 2020-07-14 21:31:52,575 - DEBUG - musicbot.config | In config.py::MainThread(140250957956928), line 251 in async_validate: Acquired owner id via API
[3.7561569213867188] 2020-07-14 21:31:52,575 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 814 in _scheck_configs: Validating permissions config
[3.7563233375549316] 2020-07-14 21:31:52,575 - DEBUG - musicbot.permissions | In permissions.py::MainThread(140250957956928), line 94 in async_validate: Validating permissions...
[3.7565703392028809] 2020-07-14 21:31:52,575 - DEBUG - musicbot.permissions | In permissions.py::MainThread(140250957956928), line 98 in async_validate: Fixing automatic owner group
[3.7567837238311768] 2020-07-14 21:31:52,575 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 979 in on_ready: Connected: 732710014216830987/Merde#6205
[3.7572257518768311] 2020-07-14 21:31:52,576 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 987 in on_ready: Owner:     193120890899267585/ٴٴ#9103

[3.7575387954711914] 2020-07-14 21:31:52,576 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 993 in on_ready: Guild List:
[3.7578756809234619] 2020-07-14 21:31:52,576 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 997 in on_ready:  - Harder
[3.7582430839538574] 2020-07-14 21:31:52,577 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 1053 in on_ready: Not bound to any text channels
[3.7584948539733887] 2020-07-14 21:31:52,577 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 1079 in on_ready: Not autojoining any voice channels
[3.7594964504241943] 2020-07-14 21:31:52,578 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 286 in _join_startup_channels: Found owner in "General"
[3.7597861289978027] 2020-07-14 21:31:52,578 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 295 in _join_startup_channels: Attempting to join Harder/General
[6.4041831493377686] 2020-07-14 21:31:55,223 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 747 in deserialize_queue: Deserializing queue for 732295950714011738
[6.6349246501922607] 2020-07-14 21:31:55,453 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 439 in get_player: Created player via deserialization for guild 732295950714011738 with 1 entries
[6.6354134082794189] 2020-07-14 21:31:55,454 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 311 in _join_startup_channels: Joined Harder/General
[6.6366314888000488] 2020-07-14 21:31:55,455 - DEBUG - musicbot.entry | In entry.py::MainThread(140250957956928), line 62 in get_ready_future: Created future for None
[6.6373357772827148] 2020-07-14 21:31:55,456 - INFO - musicbot.entry | In entry.py::MainThread(140250957956928), line 366 in _really_download: Download started: https://soundcloud.com/izanagibang/best-song-ever-created
[8.2095448970794678] 2020-07-14 21:31:57,028 - INFO - musicbot.entry | In entry.py::MainThread(140250957956928), line 376 in _really_download: Download complete: https://soundcloud.com/izanagibang/best-song-ever-created
[8.2132229804992676] 2020-07-14 21:31:57,032 - FFMPEG - musicbot.player | In player.py::MainThread(140250957956928), line 293 in _play: Creating player with options: -nostdin -vn audio_cache/soundcloud-90213207-Best_Song_Ever_Created_Maximbady_-_Hey_baby.mp3
[8.2230744361877441] 2020-07-14 21:31:57,042 - DEBUG - musicbot.player | In player.py::MainThread(140250957956928), line 306 in _play: Playing <musicbot.player.SourcePlaybackCounter object at 0x7f8eb49a2400> using <discord.voice_client.VoiceClient object at 0x7f8eb521fb50>
[8.2273652553558350] 2020-07-14 21:31:57,046 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 474 in on_player_play: Running on_player_play
[8.2278752326965332] 2020-07-14 21:31:57,046 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 723 in serialize_queue: Serializing queue for 732295950714011738
[27.0019505023956299] 2020-07-14 21:32:15,820 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 2716 in on_message: 193120890899267585/ٴٴ#9103: !skip
[27.0131967067718506] 2020-07-14 21:32:15,832 - DEBUG - musicbot.player | In player.py::Thread-3(140250671998720), line 220 in _playback_finished: Deleting file: audio_cache/soundcloud-90213207-Best_Song_Ever_Created_Maximbady_-_Hey_baby.mp3
[27.0143411159515381] 2020-07-14 21:32:15,833 - DEBUG - musicbot.player | In player.py::Thread-3(140250671998720), line 225 in _playback_finished: File deleted: audio_cache/soundcloud-90213207-Best_Song_Ever_Created_Maximbady_-_Hey_baby.mp3
[27.0178744792938232] 2020-07-14 21:32:15,836 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 550 in on_player_finished_playing: Running on_player_finished_playing
[27.2232422828674316] 2020-07-14 21:32:16,042 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 723 in serialize_queue: Serializing queue for 732295950714011738
[27.2246158123016357] 2020-07-14 21:32:16,043 - DEBUG - musicbot.bot | In bot.py::MainThread(140250957956928), line 546 in on_player_stop: Running on_player_stop
[104.0940260887145996] 2020-07-14 21:33:32,913 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 2716 in on_message: 193120890899267585/ٴٴ#9103: !stream https://youtu.be/-aLYvZ5sX28
[104.9392480850219727] 2020-07-14 21:33:33,758 - ERROR - musicbot.bot | In bot.py::MainThread(140250957956928), line 2845 in on_message: Error in stream: ExtractionError: Unknown error: [0;31mERROR:[0m Unable to download webpage: HTTP Error 429: Too Many Requests (caused by <HTTPError 429: 'Too Many Requests'>)
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/youtube_dl/extractor/common.py", line 627, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/local/lib/python3.8/dist-packages/youtube_dl/YoutubeDL.py", line 2238, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.8/urllib/request.py", line 563, in error
    result = self._call_chain(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 755, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 429: Too Many Requests

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/youtube_dl/YoutubeDL.py", line 797, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/lib/python3.8/dist-packages/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/lib/python3.8/dist-packages/youtube_dl/extractor/youtube.py", line 1782, in _real_extract
    video_webpage, urlh = self._download_webpage_handle(url, video_id)
  File "/usr/local/lib/python3.8/dist-packages/youtube_dl/extractor/youtube.py", line 276, in _download_webpage_handle
    return super(YoutubeBaseInfoExtractor, self)._download_webpage_handle(
  File "/usr/local/lib/python3.8/dist-packages/youtube_dl/extractor/common.py", line 660, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "/usr/local/lib/python3.8/dist-packages/youtube_dl/extractor/common.py", line 645, in _request_webpage
    raise ExtractorError(errmsg, sys.exc_info()[2], cause=err)
youtube_dl.utils.ExtractorError: Unable to download webpage: HTTP Error 429: Too Many Requests (caused by <HTTPError 429: 'Too Many Requests'>)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ikk157/MusicBot/musicbot/playlist.py", line 123, in add_stream_entry
    info = await self.downloader.extract_info(self.loop, song_url, download=False)
  File "/home/ikk157/MusicBot/musicbot/downloader.py", line 84, in extract_info
    return await loop.run_in_executor(self.thread_pool, functools.partial(self.unsafe_ytdl.extract_info, *args, **kwargs))
  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.8/dist-packages/youtube_dl/YoutubeDL.py", line 820, in extract_info
    self.report_error(compat_str(e), e.format_traceback())
  File "/usr/local/lib/python3.8/dist-packages/youtube_dl/YoutubeDL.py", line 625, in report_error
    self.trouble(error_message, tb)
  File "/usr/local/lib/python3.8/dist-packages/youtube_dl/YoutubeDL.py", line 595, in trouble
    raise DownloadError(message, exc_info)
youtube_dl.utils.DownloadError: [0;31mERROR:[0m Unable to download webpage: HTTP Error 429: Too Many Requests (caused by <HTTPError 429: 'Too Many Requests'>)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ikk157/MusicBot/musicbot/bot.py", line 2823, in on_message
    response = await handler(**handler_kwargs)
  File "/home/ikk157/MusicBot/musicbot/bot.py", line 1709, in cmd_stream
    await player.playlist.add_stream_entry(song_url, channel=channel, author=author)
  File "/home/ikk157/MusicBot/musicbot/playlist.py", line 138, in add_stream_entry
    raise ExtractionError("Unknown error: {}".format(e))
musicbot.exceptions.ExtractionError: Unknown error: [0;31mERROR:[0m Unable to download webpage: HTTP Error 429: Too Many Requests (caused by <HTTPError 429: 'Too Many Requests'>)
[133.1778171062469482] 2020-07-14 21:34:01,996 - INFO - musicbot.bot | In bot.py::MainThread(140250957956928), line 2716 in on_message: 193120890899267585/ٴٴ#9103: !shutdown
[133.707516432] launcher-INFO: All done. ```

**Comment by AutumnClove** [comment]
This won't fix anything about the IP block, because you'll still be downloading from the same IP. We do want to add a built in proxy in the feature to help prevent the IP block however.

**Comment by ibrahimk157** [comment]
> This won't fix anything about the IP block, because you'll still be downloading from the same IP. We do want to add a built in proxy in the feature to help prevent the IP block however.

Ah i see... I appreciate your efforts!
